main.py

```diff
@@ -0,0 +1,89 @@
+import json
+import os
+import requests
+import re
+
+
+owner, repo_name = os.environ["OWNER_AND_REPO"].split("/")
+
+url = "https://api.github.com/graphql"
+query = """
+query getDiscussions {
+    repository(name: "$repo_name", owner: "$owner") {
+        discussions(first: 100) {
+            nodes {
+                title
+                body
+                author {
+                    login
+                }
+            }
+            totalCount
+        }
+    }
+}
+"""
+
+variables = {
+    "repo_name": repo_name,
+    "owner": owner,
+}
+
+def apply_variables(query, variables):
+    for name, value in variables.items():
+        query = query.replace("$"+name, str(value))
+
+    return query
```

Comment:
You could try this:

```suggestion
def apply_variables(query, variables):
    return query.format(**{k: str(v) for k, v in variables.items()})
```

Reply:
And if we were to use built-in string formatting, we would need to escape all curly braces.

Our query would then look something like this:

```graphql
query getDiscussions {{
    repository(name: "{repo_name}", owner: "{owner}") {{
        discussions(first: 100) {{
            nodes {{
                title
                body
                author {{
                    login
                }}
            }}
            totalCount
        }}
    }}
}}
```